File: ruby/pgms_ruby/09-pgmAvionInterface/TipoComercial.rb
require_relative 'Avion'

class TipoComercial < Avion
  def initialize
    super
    self.setTipo("Comercial")
  end

  def startingEngine
    puts "⛽Check Fuel ✅"
    3.times do |i|
      print "💥💥Spark💥💥"
      begin
        sleep(1 - (i * 0.2))
      rescue Interrupt
        puts "⛔: KeyboardInterrupt"
      rescue Exception => e
        puts "⛔: #{e}"
      end
    end
    puts "Started the Motor...✈️✈️"
  end

  def stopingEngine
    puts "✈️✈️ Check Engine ✅"
    2.times do |i|
      puts "⛔ Stoping Engine...⛔‼️"
      begin
        sleep(1 - (i * 0.4))
      rescue Interrupt
        puts "⛔: KeyboardInterrupt"
      rescue Exception => e
        puts "⛔: #{e}"
      end
    end
    puts "Stoped...⛔"
  end
end

File: python/java_to_python/09_pgm_avion_interface/TipoComercial.py
import time
from Avion import Avion

class TipoComercial(Avion):
    def __init__(self):
        super().__init__()
        self.setTipo("Comercial")
    
    def startingEngine(self):
        print("⛽Check Fuel ✅")
        for i in range(3):
            print("💥💥Spark💥💥")
            try:
                time.sleep(1 - (i * 0.2))
            except Exception as e:
                print("⛔: {}".format(e))
        print("Started the Motor...✈️✈️")
    
    def stopingEngine(self):
        print("✈️✈️ Check Engine ✅")
        for i in range(2):
            print("⛔ Stoping Engine...⛔‼️")
            try:
                time.sleep(1 - (i * 0.4))
            except Exception as e:
                print("⛔: {}".format(e))
        print("Stoped...⛔")